### ISSUE TITLE
Typenames of NAMED_ENUM schemas should be derived from title when present

### ISSUE THREAD

**Issue by aparna-sath** [bcherny/json-schema-to-typescript#104]
```json
"gender": {
  "type": "string",
  "title": "GenderType",
  "tsEnumNames": ["MALE", "FEMALE"],
  "enum": [0, 1]
}
```
This should generate a typescript enum called GenderType, rather than Gender.

**Comment by bcherny** [comment]
PRs are welcome!

**Comment by bcherny** [comment]
Fixed in f9c8624.

```sh
+ json-schema-to-typescript@4.5.0
```
